pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -688,7 +681,11 @@ func (npc *NetworkPolicyController) appendRuleToPolicyChain(iptablesCmdHandler *
 		args = append(args, "-p", protocol)
 	}
 	if dPort != "" {
-		args = append(args, "--dport", dPort)
+	    if dPort == "16384" && protocol == "UDP" {
```

Comment:
Though i am not sure of the proper context. But for understanding asking this, what if the dPort was a value between 16385 - 32768, in that case also we should allow the port range right ?

Reply:
This is the code that translates netpol into iptable rules. For TE pods, we needed port-ranges to be supported; and k8s did not support port-ranges (in the netpol manifest). So we implemented a poor-man's scheme to add support for port-ranges for TE pods -- if we find a netpol that specifies port 16384/UDP; we would assume that the author of the netpol wanted to open up the entire port-range.
When this code executes kube-router will generate an IPtable rule that allows traffic for the entire portrange 16384-32768. At runtime when traffic arrives at any port value in this range; this rule will accept such traffic.

```
-A KUBE-NWPLCY-FPK44BWZQUECYIEZ -p udp -m comment --comment "rule to ACCEPT traffic from specified ipBlocks to dest pods selected by policy name: te namespace default" -m set --match-set KUBE-SRC-S5SWZCN5BKEU3P6P src -m set --match-set KUBE-DST-YUTQO226WXC5IMCE dst -m udp --dport 16384:32768 -j ACCEPT
```

Reply:
> Though i am not sure of the proper context. But for understanding asking this, what if the dPort was a value between 16385 - 32768, in that case also we should allow the port range right ?

If the dPort specified in the netpol does not match 16384, then we do not treat it is as describing a port-range. We need portRange support for TE and SBCi only for the 16384-32768 case. In every other case, ex: dport==16390 in some netpol; it will be treated as an individual port.

Reply:
So here the assumption is that always if the dPort specified in the netpol == 16384 treat that as a port-range request and we do not support allow only the individual dPort 16384 like the other ports.